src/util/Recoil_CopyOnWrite.js

```diff
@@ -25,24 +25,18 @@ function setByDeletingFromSet<V>(set: $ReadOnlySet<V>, v: V): Set<V> {
   return next;
 }
 
-function mapBySettingInMap<K, V>(
-  map: $ReadOnlyMap<K, V>,
-  k: K,
-  v: V,
-): Map<K, V> {
-  const next = new Map(map);
-  next.set(k, v);
-  return next;
+function mapBySettingInMap<K, V>(map: Map<K, V>, k: K, v: V): Map<K, V> {
```

Comment:
This is no longer "copy on write" after the change... if there's performance concern, maybe we need to re-think about how we manipulate data.